lib/multiaddr.js

```diff
@@ -3,22 +3,25 @@
  * @returns {string} Parsed URI, e.g. `http://127.0.0.1:80`
  */
 export function multiaddrToHttpUrl (addr) {
-  const [multiAddr, httpPathMultiAddr] = addr.split('/http-path/')
+  const [multiAddr, httpPathMultiAddr] = addr.split('/http-path')
   const [, hostType, hostValue, ...multiAddrParts] = multiAddr.split('/')
-  let scheme, path, rest
-  if (httpPathMultiAddr) {
+  let scheme, path, rest, port
+  if (addr.includes('/http-path')) {
     scheme = multiAddrParts.shift()
     rest = multiAddrParts
-    path = decodeURIComponent(httpPathMultiAddr)
-    if (!path) {
+    try {
+      // Remove leading slash and parse URI-encoded path
+      // If the http-path is empty both `.../http-path/` and `.../http-path` are valid
+      path = decodeURIComponent(httpPathMultiAddr.replace(/^\/+/, ''))
+    } catch (err) {
       throw Object.assign(
-        new Error(`Cannot parse "${addr}": http-path is empty`),
-        { code: 'MULTIADDR_HAS_EMPTY_HTTP_PATH' }
+        new Error(`Cannot parse "${addr}": unsupported http path`, { cause: err }),
+        { code: 'UNSUPPORTED_HTTP_PATH' }
```

Comment:
When I see the code `UNSUPPORTED_HTTP_PATH`, I expect that the multiaddr contained a valid http-path value that we (Spark) don't support.

That's not the case here, is it?

As I understand it, we want to signal that the http-path value was not a valid one.

Proposed error code:

```suggestion
        { code: 'INVALID_HTTP_PATH' }
```


Alternatively:

```suggestion
        { code: 'MALFORMED_HTTP_PATH' }
```

Reply:
Addressed in d2c6c60